src/adapter/src/coord/sequencer.rs

```diff
@@ -1836,11 +1844,24 @@ impl Coordinator {
         session: &Session,
         plan: DropRolesPlan,
     ) -> Result<ExecuteResponse, AdapterError> {
-        let ops = plan
-            .ids
-            .into_iter()
-            .map(|(id, name)| catalog::Op::DropRole { id, name })
-            .collect();
+        let dropped_ids: BTreeSet<_> = plan.ids.iter().map(|(id, _)| id).collect();
+        let mut ops = Vec::new();
+
+        // If any role is a member of a dropped role, then we must revoke that membership.
```

Comment:
What happens to object ownership if the dropped role owns something?

Reply:
Object ownership isn't implemented yet, but when we do implement it then we will add the following restriction as per the design doc (https://github.com/MaterializeInc/materialize/pull/17702/files#diff-ac93b997117d1fe5386d8624f61f1f83ae6235db01d647578791ba3c63bd5e5eR332-R335)
> We will update `DROP ROLE` so that roles cannot be dropped until it meets the following criteria:
>
> - No objects are owned by the role.
> - The role contains no privileges.

These requirements were taken directly from PostgreSQL.

Reply:
The "The role contains no privileges." requirement doesn't make a ton of sense to me tbh, but the owned one makes sense. There isn't an obvious new owner to assign the object to.

Reply:
Here's a concrete example from PostgreSQL.
![Screenshot from 2023-03-15 16-04-12](https://user-images.githubusercontent.com/10800303/225429602-13364596-ce41-4186-b37a-361702c0c8f6.png)